### PR TITLE
new test with batch norms

### PR DESCRIPTION
**Pull request by michaelsdr**
Tests the gradients with batch norms

### CODE REVIEW

File: momentumnet/tests/test_transform.py [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
 
-from torchvision.models import resnet18
+from torchvision.models import resnet101
 
 
 @pytest.mark.parametrize("use_backprop", [True, False])
 def test_resnet_vision(use_backprop):
     x = torch.randn((2, 3, 10, 10), requires_grad=True)
-    net = resnet18()
+    net = resnet101()
```
Comment (pierreablin) [inline momentumnet/tests/test_transform.py]: better parametrize the arch in the `test_resnet_vision` function like `test_resnet_vision(use_backprop, depth="18")` and test the two depths.

**Comment by pierreablin** [comment]
@michaelsdr do you know how to fix the unit tests?

File: momentumnet/tests/test_transform.py [hunk 2]
```diff
@@ -4,20 +4,29 @@
...
-    net = resnet18()
-    net(x)
+    if depth == 18:
+        net = resnet18()
+    else:
+        net = resnet101()
     momnet = transform_to_momentumnet(
         net, use_backprop=use_backprop, keep_first_layer=True
     )
-    momnet(x)
```
Comment (pierreablin) [inline momentumnet/tests/test_transform.py:20]: why remove this?

File: momentumnet/tests/test_transform.py [hunk 3]
```diff
@@ -4,20 +4,29 @@
...
 
 
-@pytest.mark.parametrize("use_backprop", [True, False])
-def test_resnet_vision(use_backprop):
+torch.manual_seed(5)
+if torch.cuda.is_available():
+    torch.cuda.manual_seed_all(5)
+
+
+@pytest.mark.parametrize(
```
Comment (pierreablin) [inline momentumnet/tests/test_transform.py:17]: can use two parametrize in series, makes it cleaner

Comment (pierreablin) [reply momentumnet/tests/test_transform.py:17]:
```
@pytest.mark.parametrize("use_backprop",  ["True", "False"])
@pytest.mark.parametrize("depth",  [18, 101])
```
